---
layout: post
section: blog
title: DPGP
date: 2022-08-02
description: Time series analysis for large omic datasets?
categories:
  - time series
  - transcriptomics
image: https://journals.plos.org/ploscompbiol/article/figure/image?size=large&id=10.1371/journal.pcbi.1005896.g003
author_staff_member:  gurkanbebek
unsplash: "9wYdW55NbnY"

---




DPGP is a Dirichlet process time series clustering tool. DPGP generates the trajectories of gene expression levels across time, where the trajectories are clustered transcriptional time series data with an infinite mixture of Gaussian processes modeled using a Gaussian process. DPGP compared to state-of-the-art time series clustering methods across a variety of simulated data performs much better. DPGP can be used for exploratory data analysis of gene expression time series to reveal novel insights into biomedically important gene regulatory processes. ([see paper](https://doi.org/10.1371/journal.pcbi.1005896)).

<meta property="og:image" content="https://unsplash.com/photos/{{ page.unsplash }}/download?w=1000">
<meta name="twitter:image" content="https://unsplash.com/photos/{{ page.unsplash }}/download?w=1000">



DPGP software is freely available on [github](https://github.com/PrincetonUniversity/DP_GP_cluster).


Here are steps to make it work (on linux).


#### 1. We will first download DPGP software from github.

```
mkdir DPGP
cd DPGP
git clone https://github.com/PrincetonUniversity/DP_GP_cluster.git
```


#### 2. Next, we need to make some modifications to make DPGP work. Unfortunately, due to python updates, the code needs fixing.

##### In DP GP cluster.py file, uncomment lines to change:

```
import matplotlib.pyplot as plt
```

#### to
```
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt
```

#### 3. DP GP cluster requires the following Python packages: GPy, pandas, numpy, scipy (>= 0.14), matplotlib.pyplot

##### ”DPGP has been tested in linux with Python 2.7 and with Anaconda distributions of the latter four above packages.” However, since 2018 things have changed. We will need to enforce versioning.


```
pip install --user GPy==1.9.8 pandas numpy scipy matplotlib cython scikit-learn


With Python 3.10:
pip install --user GPy==1.9.8 pandas numpy scipy matplotlib cython scikit-learn --use-pep517
 
```

#### 4. Finally, we can install DPGP:

```
python setup.py install --user
```

This will install /home/gxb43/.local/bin/DP_GP_cluster.py



#### 5. Tests

```
cd DP_GP_cluster
DP_GP_cluster.py -i test/test.txt -o test/test -p png -n 20 --plot
```






#### 6. plots
    
Because the optimal clustering is chosen after the entirety of Gibbs sampling, the script can be rerun with alternative clustering optimality criteria to yield different sets of clusters. Also, if `--plot` flag was not indicated when the above script is called, plots can be generated after sampling:

```
DP_GP_cluster.py \
    -i /path/to/expression.txt \
    --sim_mat /path/to/output_prefix_posterior_similarity_matrix.txt \
    --clusterings /path/to/output_prefix_clusterings.txt \
    --criterion MPEAR \
    --post_process \
    --plot --plot_types png,pdf \
    --output /path/to/output_prefix_MPEAR_optimal_clustering.txt \
    --output_path_prefix /path/to/output_prefix_MPEAR
```
 


